org-code-javabuilder/lib/src/main/java/dev/javabuilder/WebSocketServer.java

```diff
@@ -85,25 +86,37 @@ public void onOpen(Session session) {
             levelId,
             dashboardHostname,
             useNeighborhood);
-    Thread codeExecutor =
+    CodeBuilderRunnable codeBuilderRunnable =
+        new CodeBuilderRunnable(fileLoader, outputAdapter, executionType);
+    this.codeExecutor = new Thread(codeBuilderRunnable);
+    this.codeExecutor.start();
+    Thread waitToCleanup =
         new Thread(
             () -> {
-              CodeBuilderWrapper codeBuilderWrapper =
-                  new CodeBuilderWrapper(fileLoader, outputAdapter);
-              codeBuilderWrapper.executeCodeBuilder(executionType);
+              try {
+                this.codeExecutor.join();
+              } catch (InterruptedException e) {
+                // ignore this exception, it likely means the end user stopped
+                // their program.
+              }
```

Comment:
Just to make sure I understand - for the WebSocketServer case, we don't need to manually check if the connection is alive periodically because closing the websocket will automatically trigger an InterruptedException, right?